src/Types/Animations/BubblePop.tsx

```diff
@@ -1,98 +1,209 @@
-import { SketchData } from '../Figures';
+import { SelectedShape, SketchData } from '../Figures';
 import P5 from 'p5';
 import { AnimatedFigure } from '../ProcessingFigures';
 import 'p5collide';
-import { findDOMNode } from 'react-dom';
+import { newFigure } from './Animation';
 
 export class BubblePop extends Animation {
 
-    static burst(fig, sketchData) {
+    static NUM_FIGURES = 200;
+    static CACHE_LIMIT = 100;
+    static counter = 0;
 
-        if (fig.timer <= 25) {
-            let index = sketchData.figs.indexOf(fig);
-            sketchData.figs.splice(index, 1);
-            fig.pop.play();
+    static colorSettingsCache;
+    static circleCache = [];
+    static squareCache = [];
+    static triangleCache = [];
+
+    static burst_cache(fig, sketchData, p) {
+
+        let index = sketchData.figs.indexOf(fig);
+
+        let type : string = fig.getShapeDescriptor().type;
+        let dim : number = fig.dim;
+        let xpos : number = fig.pos.x;
+        let ypos : number = fig.pos.y;
+
+        for(let i=0; i < 3; ++i){
+            fig.dim += 5;
+            fig.display();
+        }
+        
+        var shape : SelectedShape;
+        var cache;
+        // var color;
+        
+        switch(type) {
+            case("CIRCLE"):
+                cache = this.circleCache;
+                shape = SelectedShape.Circle;
+                // color = getCircleColor();
```

Comment:
y no work???? 😖

Reply:
what was the error for this? my guess is that burst_cache wasn't static when you were first testing this code

Reply:
burst_cache was static. The error is: `Cannot find name 'getCircleColor'`

Reply:
`Animation.getCircleColor()`

Reply:
Idk something weird is happening. The class VSCode thinks is being extended is 
```
var Animation: {
    new (effect?: AnimationEffect, timeline?: AnimationTimeline): Animation;
    prototype: Animation;
}
```

If I try to import Animation to correct it, I get a circular dependency error because Animation also imports BubblePop

Reply:
absolute mad man vetoing review and merging himself >:

Reply:
You approved it 💁